url7a69e77641_benmouh_MySgbd_tgz-pJ8-Demo_CheckRowNumJ8/src/org/sgbd/Mysgbd/parser/MysgbdParser.java

```diff
@@ -0,0 +1,3669 @@
+      MStatement st = (MStatement)v.elementAt(i);
+      System.out.println(st.toString());
+    }
+*/
+
+    System.out.println("Parse Successful") ;
+
+  }
+
+  static final public void BasicDataTypeDeclaration() throws ParseException {
+    switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+    case K_CHAR:
+    case K_FLOAT:
+    case K_INTEGER:
+    case K_NATURAL:
+    case K_NUMBER:
+    case K_REAL:
+    case K_VARCHAR2:
+    case K_VARCHAR:
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case K_CHAR:
+        jj_consume_token(K_CHAR);
+        break;
+      case K_VARCHAR:
+        jj_consume_token(K_VARCHAR);
+        break;
+      case K_VARCHAR2:
+        jj_consume_token(K_VARCHAR2);
+        break;
+      case K_INTEGER:
+        jj_consume_token(K_INTEGER);
+        break;
+      case K_NUMBER:
+        jj_consume_token(K_NUMBER);
+        break;
+      case K_NATURAL:
+        jj_consume_token(K_NATURAL);
+        break;
+      case K_REAL:
+        jj_consume_token(K_REAL);
+        break;
+      case K_FLOAT:
+        jj_consume_token(K_FLOAT);
+        break;
+      default:
+        jj_la1[0] = jj_gen;
+        jj_consume_token(-1);
+        throw new ParseException();
+      }
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case 88:
+        jj_consume_token(88);
+        jj_consume_token(S_NUMBER);
+        switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+        case 89:
+          jj_consume_token(89);
+          jj_consume_token(S_NUMBER);
+          break;
+        default:
+          jj_la1[1] = jj_gen;
+          ;
+        }
+        jj_consume_token(90);
+        break;
+      default:
+        jj_la1[2] = jj_gen;
+        ;
+      }
+      break;
+    case K_DATE:
+      jj_consume_token(K_DATE);
+      break;
+    case K_BINARY_INTEGER:
+      jj_consume_token(K_BINARY_INTEGER);
+      break;
+    case K_BOOLEAN:
+      jj_consume_token(K_BOOLEAN);
+      break;
+    default:
+      jj_la1[3] = jj_gen;
+      jj_consume_token(-1);
+      throw new ParseException();
+    }
+  }
+
+/* ---------------- DECLARATIONS SECTION ends here ------------------ */
+
+/* ---------------- Code Section starts here ---------------------- */
+  static final public Vector SQLStatements() throws ParseException {
+  Vector v = new Vector();
+  MStatement s;
+    label_1:
+    while (true) {
+      s = SQLStatement();
+                          if(s == null) {if (true) return v;} else v.addElement(s);
+      switch ((jj_ntk==-1)?jj_ntk():jj_ntk) {
+      case K_COMMIT:
+      case K_DELETE:
+      case K_EXIT:
+      case K_INSERT:
+      case K_LOCK:
+      case K_QUIT:
+      case K_ROLLBACK:
+      case K_SELECT:
+      case K_SET:
+      case K_UPDATE:
+        ;
+        break;
+      default:
+        jj_la1[4] = jj_gen;
+        break label_1;
+      }
+    }
+      {if (true) return v;}
+    throw new Error("Missing return statement in function");
+  }
+
+  static final public MStatement SQLStatement() throws ParseException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.